misc/services/mobile-webui/mobile-webui-frontend/src/components/buttons/ButtonWithIndicator.jsx

```diff
@@ -52,10 +52,12 @@ const ButtonWithIndicator = ({
     otherProps,
   });
 
+  const consolidatedTestId = testId ?? id;
+
   return (
     <button
       id={id}
-      data-testid={testId}
+      data-testid={consolidatedTestId}
```

Comment:
hmm, not sure that's OK because 
* I want the data-testid to be as precise and as stable as possible => to avoid breaking the tests in the future
* I want to get rid of ID property for testing purposes and completely rely on data-testid

so why aren't you particularly specify the testId property?

Reply:
I've set the test id